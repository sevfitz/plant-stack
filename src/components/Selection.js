import React from 'react';
// import { connect } from 'react-redux';
import styled from 'styled-components';

const SelectButton = styled.button`
    background-color: #ffed64;
    font-size: 20px;
    height: 40px;
    width: 100px;
    border: 2px solid #ffed64;
    border-radius: 25px;
`;

export function Selection(props) {
    const { card, choice, onSelectGenus, onSelectSpecies } = props;
    const { _id, name } = card;
    
    

    return(
        <div>
            <form onSubmit={event => {
                event.preventDefault();
            }}>
                <p>I think {choice} is the {name}'s...</p>
                <SelectButton type="submit" onClick={() => onSelectGenus(_id, choice, 'genus')}>Genus</SelectButton>
                <span> OR </span>
                <SelectButton type="submit" onClick={() => onSelectSpecies(_id, choice, 'species')}>Species</SelectButton>
            </form>
        </div>
    );
}

// export default connect(
//     state => {
//         return ({ 
//             name: state.name, 
//             genus: state.genus, 
//             species: state.species, 
//             description: state.description, 
//             url: state.url,
//             seenCards: state.seenCards,
//             cards: state.cards
//     })}
// )(Selection);

// const mapStateToProps = (state) => {
//     const selectedCard = state.cards.find((card) => card._id === state.selection._id);
//     return {
//         card: selectedCard,
//         name: state.name, 
//         genus: state.genus, 
//         species: state.species, 
//         description: state.description, 
//         url: state.url,
//         seenCards: state.seenCards,
//         cards: state.cards
//     };
// };

// function mapDispatchToProps(dispatch) {
//     return bindActionCreators({ chooseGenus, chooseSpecies }, dispatch);
// }

// const mapDispatchToProps = {
//     chooseGenus,
//     chooseSpecies
// }

// export default connect(mapStateToProps, mapDispatchToProps)(Selection);
