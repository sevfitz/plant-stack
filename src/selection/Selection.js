import React from 'react';
// import { connect } from 'react-redux';
import styled from 'styled-components';

// const SelectButton = styled.button`
//     background-color: #ffed64;
//     font-size: 20px;
//     height: 40px;
//     width: 100px;
//     border: 2px solid #ffed64;
//     border-radius: 25px;
// `;

export function Selection(props) {
    const { selectedCard, selection, cards, onSelectGenus, onSelectSpecies } = props;
    const { _id, choice } = selection;
    
    

    return(
        <form className="card-footer" onSubmit={event => {
            event.preventDefault();
        }}>
            {/* <span className="card-footer-item"> */}
                <button className="button is-medium is-outlined is-primary is-pulled-left" type="submit" onClick={() => onSelectGenus(_id, choice, cards)}>Genus</button>
            {/* </span> */}
            {/* <span className="card-footer-item"></span> */}
            {/* <span className="card-footer-item"> */}
                <button className="button is-medium is-outlined is-primary is-pulled-right" type="submit" onClick={() => onSelectSpecies(_id, choice, cards)}>Species</button>
            {/* </span> */}
        </form>
    );
}

// export default connect(
//     state => {
//         return ({ 
//             name: state.name, 
//             genus: state.genus, 
//             species: state.species, 
//             description: state.description, 
//             url: state.url,
//             seenCards: state.seenCards,
//             cards: state.cards
//     })}
// )(Selection);

// const mapStateToProps = (state) => {
//     const selectedCard = state.cards.find((card) => card._id === state.selection._id);
//     return {
//         card: selectedCard,
//         name: state.name, 
//         genus: state.genus, 
//         species: state.species, 
//         description: state.description, 
//         url: state.url,
//         seenCards: state.seenCards,
//         cards: state.cards
//     };
// };

// function mapDispatchToProps(dispatch) {
//     return bindActionCreators({ chooseGenus, chooseSpecies }, dispatch);
// }

// const mapDispatchToProps = {
//     chooseGenus,
//     chooseSpecies
// }

// export default connect(mapStateToProps, mapDispatchToProps)(Selection);
