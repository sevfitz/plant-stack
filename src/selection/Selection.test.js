import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { Selection } from './Selection';

describe('Selection Component Snapshot', () => {

    it('displays correct properties', () => {
        const choice = 'some choice';
        const onSelectGenus = function() {};
        const onSelectSpecies = function() {};
        const card = {};
        const wrapper = shallow(<Selection choice onSelectGenus onSelectSpecies card/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
});
