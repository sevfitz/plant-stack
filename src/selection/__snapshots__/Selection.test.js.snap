// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selection Component Snapshot displays correct properties 1`] = `
<form
  className="card-footer"
  onSubmit={[Function]}
>
  <button
    className="button is-medium is-outlined is-primary is-pulled-left"
    onClick={[Function]}
    type="submit"
  >
    Genus
  </button>
  <button
    className="button is-medium is-outlined is-primary is-pulled-right"
    onClick={[Function]}
    type="submit"
  >
    Species
  </button>
</form>
`;
